neps/nep-0393.md

```diff
@@ -0,0 +1,516 @@
+    /// information about revoked tokens.
+    fn sbt_supply_by_class(&self, ctr: AccountId, class: ClassId) -> u64;
+
+    /// Returns total supply of SBTs for a given owner. See `sbt_supply` for information about
+    /// revoked tokens.
+    /// If class is specified, returns only owner supply of the given class -- must be 0 or 1.
+    fn sbt_supply_by_owner(
+        &self,
+        account: AccountId,
+        ctr: AccountId,
+        class: Option<ClassId>,
+    ) -> u64;
+
+    /// Query sbt tokens issued by a given contract.
+    /// If `from_token` is not specified, then `from_token` should be assumed
+    /// to be the first valid token id.
+    fn sbt_tokens(&self, ctr: AccountId, from_token: Option<u64>, limit: Option<u32>)
+        -> Vec<Token>;
+
+    /// Query SBT tokens by owner
+    /// If `from_class` is not specified, then `from_class` should be assumed to be the first
+    /// valid class id.
+    /// Returns list of pairs: `(Contract address, list of token IDs)`.
+    fn sbt_tokens_by_owner(
+        &self,
+        account: AccountId,
+        ctr: Option<AccountId>,
+        from_class: Option<u64>,
+        limit: Option<u32>,
+    ) -> Vec<(AccountId, Vec<OwnedToken>)>;
+
+    /// checks if an `account` was banned by the registry.
+    fn is_banned(&self, account: AccountId) -> bool;
+
+    /*************
+     * Transactions
+     *************/
+
+    /// Creates a new, unique token and assigns it to the `receiver`.
+    /// `token_spec` is a vector of pairs: owner AccountId and TokenMetadata.
+    /// Each TokenMetadata must have non zero `class`.
+    /// Must be called by an SBT contract.
+    /// Must emit `Mint` event.
+    /// Must provide enough NEAR to cover registry storage cost.
+    // #[payable]
+    fn sbt_mint(&mut self, token_spec: Vec<(AccountId, Vec<TokenMetadata>)>) -> Vec<TokenId>;
+
+    /// sbt_recover reassigns all tokens from the old owner to a new owner,
+    /// and registers `old_owner` to a burned addresses registry.
+    /// Must be called by an SBT contract.
+    /// Must emit `Recover` event.
+    /// Must be called by an operator.
+    /// Must provide enough NEAR to cover registry storage cost.
+    /// Requires attaching enough tokens to cover the storage growth.
+    // #[payable]
+    fn sbt_recover(&mut self, from: AccountId, to: AccountId);
+
+    /// sbt_renew will update the expire time of provided tokens.
+    /// `expires_at` is a unix timestamp (in miliseconds).
+    /// Must be called by an SBT contract.
+    /// Must emit `Renew` event.
+    fn sbt_renew(&mut self, tokens: Vec<TokenId>, expires_at: u64);
+
+    /// Revokes SBT, could potentially burn it or update the expire time.
+    /// Must be called by an SBT contract.
+    /// Must emit one of `Revoke` or `Burn` event.
+    /// Returns true if a token is a valid, active SBT. Otherwise returns false.
+    fn sbt_revoke(&mut self, token: u64) -> bool;
```

Comment:
It seems this adds very little value given that `sbt_burn` also exists. I suggest removing this from the standard, as it is completely redundant. If a token exists, but should not be displayed, it should just not exist.

Reply:
`sbt_revoke` and `sbt_burn` are very different. As documented, the former is called by the issuer, the latter by an owner. With `sbt_revoke` an issuer may revoke previously assigned SBT if it finds out that there was an abuse or a scam.

Reply:
I still see this as two things that achieve the same thing. Why not just let the issuer burn tokens as well?

Also, I would suggest making it clearer which methods are meant to be called by issuers, and by the user (if you insist on sticking with the flawed registry concept).

Reply:
This might be a need idea to semantically distinguish methods expected to be called by soul owners, from all the other API methods that are generally expected to be called by sbt emitter contracts.

Would having a prefix (or suffix) help (e.g `sbt_owner_burn`) or does a note in the method docs suffice?